lib/cn/kvset.c

```diff
@@ -682,7 +682,7 @@ kvset_open2(
              * who might otherwise try to access it.
              */
 #if defined(HSE_BUILD_DEBUG) && !defined(NDEBUG)
-            mprotect(kb->kb_kblk_desc.map_base, PAGE_SIZE, PROT_NONE);
+            mprotect((void *)kb->kb_kblk_desc.map_base, PAGE_SIZE, PROT_NONE);
```

Comment:
Why do we only mprotect if kcachesize is greater than 0

Reply:
Maybe because if we don't cache them then we actually have to touch those pages at runtime?

Reply:
@beckerg, based on your about casting away const, this too is a red flag and should be avoided.  How can it be avoided in this case?  I don't want to make `map_base` a `void *` -- because that's the whole point of this PR.

Reply:
Agreed, just delete it all, any interface that does an mprotect or madvise should probably just accept an mcache map where the ptr is the same as what you get back from mmap...

Reply:
There are only two mprotect/madvise calls and one of them is only for debugging. Is it really that hard to reason about?

Reply:
I want to get rid of the mcache map abstraction and have CN use:
```
merr_t
mpool_mblock_mmap(struct mpool *mp, uint64_t mbid, const void **addr_out);

merr_t
mpool_mblock_munmap(struct mpool *mp, uint64_t mbid);
```

Then have CN use `madvise` and `mprotect` directly.  Which will put us right back here:  needing to cast away const.